### PR TITLE
expose log determinant spd

### PR DESCRIPTION
**Pull request by spinkney**
#### Submission Checklist
- [x] Fixes https://github.com/stan-dev/math/issues/2679
- [ ] Run unit tests
- Documentation
    - [x] If a user-facing facing change was made, the documentation PR is here: [<LINK>](https://github.com/stan-dev/docs/pull/520)

## Release notes

Added `log_determinant_spd` which is faster when the input matrix is known to be symmetric and positive definite.

## Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the BSD 3-clause license (https://opensource.org/licenses/BSD-3-Clause)

### CODE REVIEW

File: src/middle/Stan_math_signatures.ml
Comment (WardBrian) [inline src/middle/Stan_math_signatures.ml:1595]: Why did `log_determinant` change from SoA to AoS?

Comment (spinkney) [reply src/middle/Stan_math_signatures.ml:1595]: I didn't mean to touch that and don't recall doing anything to it. I can revert that. What do those mean?

Comment (WardBrian) [reply src/middle/Stan_math_signatures.ml:1595]: They indicate whether or not the function arguments can be Struct of Arrays matrices when the compiler is performing optimizations. If the new `log_determinant_spd` has the correct templates to be used with a `var_matrix` it can also be marked as SoA

**Comment by rok-cesnovar** [comment]
The tests now pass with the math PR branch.

The PR looks good, just needs a docs PR with some basic function docs and this should be good to go.

**Comment by spinkney** [comment]
@rok-cesnovar should I change to SoA or keep it as AoS?

**Comment by spinkney** [comment]
Also the doc PR is https://github.com/stan-dev/docs/pull/520

**Comment by rok-cesnovar** [comment]
Great! If the function supports “varmat”, lets mark it SoA, else we keep it AoS.

**Comment by rok-cesnovar** [comment]
Mark it as SoA and we can let Jenkins tell us :)

**Comment by rok-cesnovar** [comment]
The Jenkinsfile isn't set up for this case, so it didn't start the required tests, but I verified locally that varmat is supported. 

The PR looks great, I will officially approve once the Math PR is merged.

**Comment by WardBrian** [comment]
This is dependent on https://github.com/stan-dev/math/pull/2719, correct?

**Comment by spinkney** [comment]
> This is dependent on [stan-dev/math#2719](https://github.com/stan-dev/math/pull/2719), correct?

yes and I believe I've addressed @SteveBronder's review. Once he approves #2719 I'll tag @WardBrian here to review.

**Review by rok-cesnovar — APPROVED** [review]
The Math PR was merged, restarted tests - assuming they pass this is good to go! 🚀

**Pull request merged**
